### ISSUE TITLE
RN 0.54 / React 16.3 deprecation warnings for componentWillReceiveProps, componentWillUpdate

### ISSUE THREAD

**Issue by peacechen** [archriss/react-native-render-html#135]
```
Warning: componentWillReceiveProps is deprecated and will be removed in the next major version. Use static getDerivedStateFromProps instead.

Please update the following components: HTML
Learn more about this warning here:
https://fb.me/react-async-component-lifecycle-hooks

Warning: componentWillMount is deprecated and will be removed in the next major version. Use componentDidUpdate instead. As a temporary workaround, you can rename to UNSAFE_componentWillUpdate.

Please update the following components: HTML
```

The fixes are actually simple and maintain backward compatibility.
Replace `componentWillMount` with `componentDidMount`.
Replace `componentWillReceiveProps` with `componentDidUpdate` and tweak logic appropriately.

In the rare case that `getDerivedStateFromProps` is necessary, a polyfill can be used to maintain backwards compatibility.
See https://github.com/reactjs/react-lifecycles-compat

### Is this a bug report or a feature request?

Bug

### Have you read [the guidelines](https://github.com/archriss/react-native-render-html/blob/master/CONTRIBUTING.md) regarding bug report?

Yes

### Have you read [the documentation](https://github.com/archriss/react-native-render-html/blob/master/README.md) in its entirety?

Yes

### Have you made sure that your issue hasn't already been reported/solved?

Yes

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

Both

### Is the bug reproductible in a production environment (not a debug one)?

Debug (deprecation warning; production will break in future versions of RN / React)

### Have you been able to reproduce the bug in [the provided example](https://github.com/archriss/react-native-render-html/tree/master/Demo)?

Yes

### Environment

  Environment:
    React: 16.3.1
    React native: 0.55.1
    react-native-render-html: 3.9.0

  Target Platform:
    Android (6.0)
    iOS (10.3)

**Comment by peacechen** [comment]
Created PR #136 

Please review the logic change in `componentDidUpdate`.  HTML.js had both `componentWillReceiveProps ` and `componentDidUpdate`.  They have been merged into `componentDidUpdate`

**Comment by bd-arc** [comment]
Thank you for the PR! It has finally been merged and is available in version `4.2.0`.
